File: OOP_C++/tema5/lab5_1.cpp
#include<iostream>
#include<cmath>
using namespace std;

class Volume{
public:
  void volume(float l){
      cout<<"Volumul cubului este: "<<l*l*l<<endl;
    }

    void volume(float r,string esteSfera) {
      if (esteSfera=="DA")
        cout<<"Volumul sferei este: "<<(M_PI*r*r*r)*4/3<<endl;
      else cout<<"Nu este sfera"<<endl;

  }

    void volume(float Ariab,float h) {
      cout<<"Volumul piramidei este: "<<(Ariab*h)/3<<endl;
  }

  void volume() {
    cout<<"Introduceti parametrii"<<endl;
  }

};

int main(){
Volume v1;
  v1.volume();
  v1.volume(12.44);
  v1.volume(10.22,"DA");
  v1.volume(10.22,20.22);
  return 0;
}

File: OOP_Kotlin/tema5/lab5_1.kt
import kotlin.math.PI

class Volume{

    fun volume(l: Float): Float{
        return l*l*l

    }
    fun volume(r: Float, esteSfera: String ): Float{

        if(esteSfera == "DA")
            return ((r*r*r*PI)*4/3).toFloat()
         else {
             return 0f
         }

    }

    fun volume(Ariab: Float, h: Float): Float{
        return (Ariab*h)/2
    }

    fun volume(){
        println("Introduceti parametrii")
    }
}

fun main(){
    val volume = Volume()
    volume.volume()
    println(volume.volume(12.44f))
    println(volume.volume(10.22f,"DA"))
    println(volume.volume(10.22f, 20.22f))
}